opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexAnalysisSettings3.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018-2019, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.index;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a serializable gathering of some top-level metadata concerning the
+ * operation of {@link IndexDatabase} -- and persisted therein too -- which are
+ * re-compared upon each indexing run since changes to them might require
+ * re-indexing particular files or in certain cases all files.
+ */
+public final class IndexAnalysisSettings3 implements Serializable {
```

Comment:
why 3?

Reply:
First there was `IndexAnalysisSettings` with `INDEX_ANALYSIS_SETTINGS_OBJVER` set to 1. Then `IndexAnalysisSettings2` with version set to 2. Then someone renamed `IndexAnalysisSettings2` to `IndexAnalysisSettings`. Now this patch has `IndexAnalysisSettings3` with version set to 3.